api/historydata.php

```diff
@@ -37,27 +30,130 @@ function fail($s) {
 
 if(isset($_REQUEST['platform'])) {
   $platform = $_REQUEST['platform'];
-  if(!preg_match('/^('.implode('|', $allowed_platforms).')$/', $platform)) {
-    fail('Invalid platform: Only '.implode('/', $allowed_platforms).' allowed');
+  if (!preg_match('/^(' . implode('|', $allowed_platforms) . ')$/', $platform)) {
+    fail('Invalid platform: Only ' . implode('/', $allowed_platforms) . ' allowed');
   }
-} else {
+} else if ($version == '2.0') {
+  $platform = 'all';
+} else if ($version == '1.0') {
```

Comment:
```suggestion
} else //if ($version == '1.0') {
```

While it will have the same result currently, future maintenance may break this if we are not careful otherwise...